app/screens/welcome/final_screen.tsx

```diff
@@ -50,6 +51,7 @@ export default function WelcomeFinalScreen({ navigation, firestoreCtrl }: any) {
               index: 0,
               routes: [{ name: "Home" }],
             });
+            auth.signOut();
```

Comment:
Is that really necessary ?

Reply:
Yes actually. If you had recently logged in and you go as guest, you were still logged in which doesn't make any sense